### PR TITLE
Add honeybadger overrides for cloud

### PR DESCRIPTION
**Pull request by bthesorceror**

### CODE REVIEW

File: ui/src/cloud/utils/overrides.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import _ from 'lodash'
+import HoneyBadger from 'honeybadger-js'
+import {GIT_SHA} from 'src/shared/constants'
+
+if (process.env.CLOUD === 'true') {
```
Comment (OfTheDelmer) [inline ui/src/cloud/utils/overrides.ts]: Configure accepts a disabled true/false key where this value could be used.

File: ui/src/cloud/utils/overrides.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  HoneyBadger.configure({
+    apiKey: process.env.HONEYBADGER_KEY,
+    revision: GIT_SHA,
+    environment: process.env.HONEYBADGER_ENV,
```
Comment (OfTheDelmer) [inline ui/src/cloud/utils/overrides.ts]: This accepts a context where we could add more helpful information like user_id. Or in the honeybadger client we could set context using the honeybadger.resetContext or .addContext

File: ui/src/cloud/utils/overrides.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  })
+
+  const override = () => {
+    if (!window || !window.console) {
+      return
+    }
+
+    const logError = console.error
+
+    console.error = function (...args) {
```
Comment (OfTheDelmer) [inline ui/src/cloud/utils/overrides.ts]: Nice strategy

File: ui/src/cloud/utils/overrides.ts (hunk 1 above)
Comment (bthesorceror) [reply ui/src/cloud/utils/overrides.ts]: I would prefer we didn't even hit configure when its not running in cloud.

**Pull request merged**